hydra_base/db/model.py

```diff
@@ -1298,6 +1300,65 @@ def add_resourcegroup_item(self, ref_key, resource, group_id):
             group_item_i.link     = resource
         self.resourcegroupitems.append(group_item_i)
 
+    def get_data(self, child_data=None, get_parent_data=True):
+        """
+            Return all the resourcescenarios relevant to this scenario.
+            If this scenario inherits from another, look up the tree to compile
+            an exhaustive list of resourcescnearios, removing any duplicates, prioritising
+            the ones closest to this scenario (my immediate parent's values are used instead
+            of its parents)
+        """
+        
+        #This avoids python's mutable keyword argumets causing child_data to keep its values beween
+        #function calls
+        if child_data is None:
+            child_data = []
+
+        #Idenify all existing resource attr ids, which take priority over anything in here
+        childrens_ras = []
+        for child_rs in child_data:
+            childrens_ras.append(child_rs.resource_attr_id)
+        
+        #Add resource attributes which are not defined already 
+        for this_rs in self.resourcescenarios:
+            if this_rs.resource_attr_id not in childrens_ras:
+                child_data.append(this_rs)
+
+        if self.parent is not None and get_parent_data is True:
+            return self.parent.get_data(child_data)
```

Comment:
Does this need to pass on `get_parent_data=get_parent_data`. Currently it will only do 1 recursion?

Reply:
done, with test included